File: 백준/Silver/10773. 제로/제로.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val k = readLine().toInt()
    val stack = Stack<Int>()
    repeat(k) {
        val num = readLine().toInt()
        when (num) {
            0 -> {
                if (stack.isNotEmpty()) {
                    stack.pop()
                }
            }

            else -> {
                stack.push(num)
            }
        }
    }
    var sum = 0
    for (i in 0 until stack.size) {
        sum += stack[i]
    }
    println(sum)
}

File: 백준/Silver/10773. 제로/제로.py
import sys

stack = []

def main():
    input = sys.stdin.readline
    k = int(input().strip())
    stack = []

    for _ in range(k):
        num = int(input().strip())
        if num == 0:
            stack.pop()
        else:
            stack.append(num)

    sys.stdout.write(str(sum(stack)) + "\n")

if __name__ == "__main__":
    main()